lib/transform.js

```diff
@@ -64,7 +64,7 @@ TreeTransformer.prototype = new TreeWalker;
                     x = this;
                     descend(x, tw);
                 } else {
-                    tw.stack[tw.stack.length - 1] = x = this.clone();
+                    tw.stack[tw.stack.length - 1] = x = this;
                     descend(x, tw);
```

Comment:
I'm pretty sure I had a reason to `clone()` here, why did you remove it?

Reply:
> I removed it because the code doesn't work with clone. No one was using the after callback

The clone was assigned into `x` and then passed both into `after` and `descend`.

I understand current behavior is more robust and easier to work with, as cloning during transformation makes it hard to compare nodes (one can sometimes unknowingly compare cloned and original versions of the node).

The documentation at http://lisperator.net/uglifyjs/transform needs changing, as it states:

> * if an after visitor exists, then the tree walker _will clone the current node_, descend into it

This is no loner true after this change.